src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11349,165 +11355,263 @@ private ExpressionSyntax ParseIsExpression(ExpressionSyntax leftOperand, SyntaxT
             };
         }
 
-        private ExpressionSyntax ParseTerm(Precedence precedence)
-            => this.ParsePostFixExpression(ParseTermWithoutPostfix(precedence));
-
-        private ExpressionSyntax ParseTermWithoutPostfix(Precedence precedence)
+        private ExpressionSyntax ParsePrimaryExpression(Precedence precedence)
         {
-            var tk = this.CurrentToken.Kind;
-            switch (tk)
+            // Primary expressions:
+            // x.y, f(x), a[i], x?.y, x?[y], x++, x--, x!, new, typeof, checked, unchecked, default, nameof, delegate, sizeof, stackalloc, x->y
+            //
+            // Note that postfix operators (like ++) are still primary expressions, even though their prefix equivalents (`++x`) are unary.
+
+            return parsePostFixExpression(parsePrimaryExpressionWithoutPostfix(precedence));
```

Comment:
Inlined these methods since no one else shoudl call them.  now it's clearer that parsing a primary is just prsing the initial part, and the the postfix portion.